presto-product-tests/src/main/java/io/prestosql/tests/hive/ACIDTestHelper.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.tests.hive;
+
+import com.google.common.net.HostAndPort;
+import io.prestosql.plugin.hive.metastore.thrift.NoHiveMetastoreAuthentication;
+import io.prestosql.plugin.hive.metastore.thrift.ThriftHiveMetastoreClient;
+import io.prestosql.plugin.hive.metastore.thrift.Transport;
+import io.prestosql.tempto.context.ThreadLocalTestContextHolder;
+import io.prestosql.tempto.query.QueryExecutor;
+import org.apache.hadoop.hive.metastore.api.AllocateTableWriteIdsRequest;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.hadoop.hive.ql.io.AcidUtils;
+import org.apache.thrift.TException;
+
+import java.net.URI;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.UUID;
+
+import static io.prestosql.tempto.assertions.QueryAssert.Row.row;
+import static io.prestosql.tempto.assertions.QueryAssert.assertThat;
+
+public final class ACIDTestHelper
+{
+    private static final String LOCALHOST = "localhost";
+
+    private ACIDTestHelper()
+    {
+    }
+
+    // This simulates a aborted transaction which leaves behind a file in a partition of a table as follows:
+    // 1. Open Txn
+    // 2. Rollback Txn
+    // 3. Create a new table with location as the given partition location + delta directory for the rolled back transaction
```

Comment:
Does it do what we want?
You are allocating write id for table A. And then you create table B for which the writeId for the given transaction was not allocated at all.
There is no relation between table B and writeId allocated so it does not seem to simulate situation that transaction for table B was aborted, and files were left behind.

Reply:
Table A is a transactional table. It insertes some data say X, Y. And then I created a transaction and then aborted it which creates a delta.

Then, I am creating an external table B (non transactional) on the aborted transaction delta and then inserting some data (say Z) getting stored on this location. **This step is to simulate that the aborted transaction has left some data at the location.** 

Then, in the main test, I am querying Table A(transactional table), and checking that the aborted transaction delta is not getting read.

Reply:
Cool - I get it now.